core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -311,6 +315,7 @@ public String toString() {
     Preconditions.checkArgument(
         metadataFileLocation == null || changes.isEmpty(),
         "Cannot create TableMetadata with a metadata location and changes");
+    Preconditions.checkArgument(keys != null, "Keys cannot be null");
```

Comment:
maybe update the error msg to mention that it's about `Encryption Keys` as `Keys` might not be immediately obvious